addons/account/models/account.py

```diff
@@ -690,50 +691,91 @@ def compute_all(self, price_unit, currency=None, quantity=1.0, product=None, par
         if not round_tax:
             prec += 5
 
-        base_values = self.env.context.get('base_values')
-        if not base_values:
-            total_excluded = total_included = base = round(price_unit * quantity, prec)
-        else:
-            total_excluded, total_included, base = base_values
-
         # Sorting key is mandatory in this case. When no key is provided, sorted() will perform a
         # search. However, the search method is overridden in account.tax in order to add a domain
         # depending on the context. This domain might filter out some taxes from self, e.g. in the
         # case of group taxes.
-        for tax in self.sorted(key=lambda r: r.sequence):
+        sorted_taxes = self.sorted(key=lambda r: r.sequence)
+
+        base_value = self.env.context.get('base_value')
+        if base_value:
+            base = base_value
+        else:
+            base = round(price_unit * quantity, prec)
+
+        # For the computation of move lines, we could have a negative base value.
+        # In this case, compute all with positive values and negative them at the end.
+        if base < 0:
+            base = -base
+            sign = -1
+        else:
+            sign = 1
```

Comment:
Suppose you have a move line of -200 to be reported to the credit with a fixed tax of 10, price_include=True, include_base_amount=True.

Starting with -200, the computation will be -200-10 = -210 instead of - (200 - 10) = -190.

see `test_tax_move_lines_creation` in test_tax.py